### PR TITLE
Return DiaSymReader library for cross-arch crossgen to Runtime package

### PR DESCRIPTION
**Pull request by AntonLapounov**
Merge #50149 to preview3 branch.

### Customer Impact

The `Microsoft.NETCore.App.Runtime.win-arm` package misses the `Microsoft.DiaSymReader.Native.x86.dll` library required by the x86-hosted `crossgen.exe` in order to generate PDBs in the cross-architecture compilation scenario. Similarly, the `Microsoft.NETCore.App.Runtime.win-arm64` package misses the `Microsoft.DiaSymReader.Native.amd64.dll` library required by the x64-hosted `crossgen.exe`. This blocks integration to AspNetCore repo.

### Testing

The fix has been tested in the main branches of Runtime and AspNetCore repos.

### Risk

No known risk.

### CODE REVIEW

**Review by trylek — APPROVED** [review]
LGTM, thanks for fixing this!

**Comment by AntonLapounov** [comment]
@jeffschwMSFT Who can authorize this PR to unblock aspnetcore?

**Comment by jeffschwMSFT** [comment]
Anyone should be able to.  Adding @Anipik if there is any permissions issues.  Do you have a gist/pull with the necessary changes?

**Comment by AntonLapounov** [comment]
I am seeing "Merging is blocked. The base branch restricts merging to authorized users." for this PR.

**Comment by AntonLapounov** [comment]
@Anipik Could you please merge this?

**Comment by jeffschwMSFT** [comment]
Adding @ViktorHofer as @Anipik is out of office.  Who has merge permissions in preview3?

**Comment by dougbu** [comment]
@wtgodbe do you have preview branch push rights in this repo❔

**Comment by wtgodbe** [comment]
> do you have preview branch push rights in this repo❔

I do not, though I probably should in case I ever need to cover for a 5.0/6.0 build - CC @mmitche who probably does have permissions

**Comment by dougbu** [comment]
@mmitche @jeffschwMSFT @jaredpar any other ideas who has admin rights in this repo❔ Or, is this waiting for an ask-mode dance❔

**Comment by jeffschwMSFT** [comment]
Apparently I do, and I just merged.

**Comment by dougbu** [comment]
Much appreciated. This should unblock dotnet/aspnetcore#31255 🎉

**Pull request merged**
